test/types/string/psahabu/perf/allocate.chpl

```diff
@@ -0,0 +1,45 @@
+use Time;
+
+config const n = 10000000;
+config const timing = true;
+
+// concatenate
+var tConcat: Timer;
+if timing then tConcat.start();
+var concat: string;
+for i in 1..n do
+  concat = "concat is short for " + "concatenation";
+if timing then tConcat.stop();
+
+// append
+var tAppend: Timer;
+if timing then tAppend.start();
+var append: string;
+for i in 1..n do
+  append += "append";
+if timing then tAppend.stop();
+
+// copy
+var tCopy: Timer;
+if timing then tCopy.start();
+var copied = "copycat";
+for i in 1..n do
+  var copy = new string(copied);
```

Comment:
I think we should use 'i' somehow so that the body doesn't get hoisted or dead code eliminated.